core/coreunix/add.go

```diff
@@ -22,8 +22,12 @@ import (
 
 var log = eventlog.Logger("coreunix")
 
-// Add builds a merkledag from the a reader, pinning all objects to the local
-// datastore. Returns a key representing the root node.
+// DEPRECATED: Add builds a merkledag from the a reader, pinning all
+// objects to the local datastore. Returns a key representing the root
+// node.
+//
+// This function is deprecated and will be removed in 0.4.0.  You
+// should use shell/unixfs's AddFromReader instead.
```

Comment:
On Fri, May 01, 2015 at 04:58:19AM -0700, Juan Batiz-Benet wrote:
> > -// Add builds a merkledag from the a reader, pinning all objects to the local
> > -// datastore. Returns a key representing the root node.
> > +// DEPRECATED: Add builds a merkledag from the a reader, pinning all
> > +// objects to the local datastore. Returns a key representing the root
> > +// node.
> > +//
> > +// This function is deprecated and will be removed in 0.4.0.  You
> > +// should use shell/unixfs's AddFromReader instead.
> 
> please remove this comment. regardless of the future, today, this
> **is not** deprecated.

Most of my later commits in this branch (and some of the earlier ones)
are responding to this deprecation.  I've shuffled things around a
bit, so I think you should be able to merge the following initial
commits without controversy:

* 0ce5784 cmd/ipfs/init: Replace coreunix.Add with importer.BuildDagFromReader
* bf9b9fc cmd/ipfswatch/main: Replace coreunix.Add with coreunix.AddR
* ac0a606 cmd/ipfs/init: bytes.NewBufferString -> strings.NewReader

Once we've resolved strings vs []bytes [1], we can merge:

* 7f62410 util/testutil/addcat/addcat: Factor out add/cat/verify helper
* ac0a606 cmd/ipfs/init: bytes.NewBufferString -> strings.NewReader

And then once we've decided to deprecate coreunix.Add we can merge:

* cd58a84 shell/unixfs: Add a high-level filesystem-node package
* 56807e9 core/coreunix/add: Deprecate Add
* 2409fee util/testutil/addcat: importer.BuildDagFromReader -> unixfs.AddFromReader
* 159783f test/supernode_client/main.go: Migrate to unixfs.AddFromReader
* 5d97c8c core/corehttp/gateway_test: Replace coreunix.Add with importer.BuildDagFromReader
* 8a4b861 core/corehttp/gateway_test: Externalize test to use shell/unixfs' AddFromReader

Until we want to deprecate coreunix.Add, the shell/unixfs stuff is
just going to add another interface to maintain (instead of migrating
us to a new interface), so I think it's just going to cause confusion.
There's no rush though, so I'm happy to just let those commits cook
until we are ready to axe coreunix.Add.

If that sounds reasonable, I'm fine with you just merging up to the
point you're comfortable with from this branch.  I'm also happy
opening new PRs for the non-controversial changes, or pushing the
controversial changes out into new PRs.  Whatever works best for you.

[1]: https://github.com/ipfs/go-ipfs/pull/1136/files#r29509838